rosetta-source/src/main/rosetta/base-staticdata-asset-common-type.rosetta

```diff
@@ -5,11 +5,74 @@ import cdm.base.*
 
+// ART-01 
+// new Data Types:  AssetIdentifier, AssetBase, Asset, Cash, Derivative, DigitalAsset.
+// Added/changed inheritance from ProductBase to AssetBase for Assets that have an Identifier:  AssetPool, DigitalAsset, Commodity, Loan, Security.
+// Removed inheritance from ProductBase:  Index.
+// Marked as deprecated:  IdentifiedProduct.
+// Added ProductTaxonomy as an attribute to Assets that require it:  Commodity, Security.
+// Added Issuer to Security.
+//
+// Outstanding:
+// 1. Is AssetPool required (and is it an Asset?).
+// 2. Collapse Loan into Security and add Loan as a value of SecurityType.
+// 3. Use of UNION for Asset.
+// 4. Modeling of attributes of Digital Asset.
+// 5. Move Index to the right namespace.
+// 6. Metadata addressing for AssetIdentifier.
+
+type AssetIdentifier extends Identifier: <"As Asset Identifier is used to designate a particular Asset, using the Identifier construct as-is."> 
+    identifierType ProductIdTypeEnum (1..1)  <"Defines the symbology source of the Product Identifier, eg CUSIP, ISIN, etc.">
+
+type AssetBase:  <"An abstract class for common attributes for an Asset, including an Identifier.">
+    identifier AssetIdentifier (0..1)  
+
+type Asset:  <"An Asset is defined something that can be held and transferred in the financial markets.">
+    [rootType]
+    assetPool AssetPool (0..1) <"An Asset that consists of a pool of other assets that back an asset backed security.">
+    basket Basket (0..1)  <"An Asset that is a collection of other Assets.">
+    cash Cash (0..1) <"An Asset that consists solely of a monetary holding in a currency.">
+    commodity Commodity (0..1)  <"An Asset comprised of raw or refined materials or agricultural products, eg gold, oil or wheat.">
+    derivative Derivative (0..1)  <"An Asset whose value is dependent on the value of another Asset.">
+    digitalAsset DigitalAsset (0..1)  <"An Asset that exists only in digital form, eg Bitcoin or Ethereum; excludes the digital representation of other Assets.">
+    loan Loan (0..1) <"An Asset that represents a loan or borrow obligation.">
+    security Security (0..1) <"A specific type of Asset that is issued by a party to be held by or transferred to others.">
+    condition OneOfAsset:  <"An Asset must be of one and only one of the following.">
+        required choice assetPool, basket, cash, commodity, derivative, digitalAsset, loan, security
+// Using the 'required choice' condition, rather than 'one-of' as it allows other shared attributes to be added to the data type.
+
+// To improve the model, it is proposed to replace this design with a new Union construct.  The revised modelling:
+//
+// union Asset:  <"An Asset is defined something that can be held and transferred in the financial markets.">
+//      AssetPool       <"An Asset that consists of a pool of other assets that back an asset backed security.">
+//      Basket          <"An Asset that is a collection of other Assets.">
```

Comment:
**DISAGREE** 
- with Basket description that is saying **Basket Components** shall be of type **Asset**

**PROPOSAL** 
- **Basket Components** shall be of type **Observable**

**RATIONALE** 
cumulative reasons below :
  - Basket of CurrencyPairs may exist (e.g. Currency CorrelationSwap)
  - as well as a Basket of Index (less "vanilla" occurrence, but may exist in some Structured Product) 
  - more generally, there is eventually no theoretical reason no to structure a payout that would observe a Basket of **any** mix...

Reply:
Your proposals will be included in the next version of the Open Contribution (and in the Task Force meeting on 11 April) with designation "ART-04".